File: 075_Sort_Colors/075_1.c
void swap(int *nums, int i, int j) {
    int tmp = nums[i];
    nums[i] = nums[j];
    nums[j] = tmp;
}

void sortColors(int* nums, int numsSize){
    if (numsSize == 0) return nums;
    int i = 0, left = 0, right = numsSize;
    while (i < right) {
        if (nums[i] == 0) {
            swap(nums, i, left);
            left++;
            i++;
        } else if (nums[i] == 2) {
            --right;
            swap(nums, i, right);
        } else {
            i++;
        }
    }
    return nums;
}

File: 075_Sort_Colors/075_1.py
from typing import List

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        if not nums:
            return

        left, right = 0, len(nums)
        i = 0
        while i < right:
            if nums[i] == 0:
                nums[left], nums[i] = nums[i], nums[left]
                left += 1
                i += 1
            elif nums[i] == 2:
                right -= 1
                nums[right], nums[i] = nums[i], nums[right]
            else:
                i += 1

        return nums